charts/prometheus/templates/server/deploy.yaml

```diff
@@ -243,5 +234,20 @@ spec:
             optional: {{ . }}
             {{- end }}
       {{- end }}
+{{- if .Values.server.extraVolumes }}
+{{ toYaml .Values.server.extraVolumes | indent 8}}
+{{- end }}
+        - name: storage-volume
+        {{- if .Values.server.persistentVolume.enabled }}
+          persistentVolumeClaim:
+            claimName: {{ if .Values.server.persistentVolume.existingClaim }}{{ .Values.server.persistentVolume.existingClaim }}{{- else }}{{ template "prometheus.server.fullname" . }}{{- end }}
+        {{- else }}
+          emptyDir:
+          {{- if .Values.server.emptyDir.sizeLimit }}
+            sizeLimit: {{ .Values.server.emptyDir.sizeLimit }}
+          {{- else }}
+            {}
+          {{- end -}}
+        {{- end -}}
```

Comment:
These moves are not necessary.

Reply:
Compare the two files (`sts.yaml` and `deploy.yaml`) where problem sits. As stated in comment, another option is to move most parts to a named template so no more alignment/sync issues between both. Or a single with several `if` :thinking:  ?